src/DynamoCore/Models/DynamoModelEvents.cs

```diff
@@ -105,6 +106,28 @@ public virtual void OnCleanup()
                 CleaningUp(this);
         }
 
+        private void OnNodeCollectionChanged(object sender,
+            NotifyCollectionChangedEventArgs e)
+        {
+            switch (e.Action)
+            {
+                case NotifyCollectionChangedAction.Add:
+                    foreach (NodeModel newItem in e.NewItems)
+                        newItem.PropertyChanged += OnNodePropertyChanged;
+                    break;
+
+                case NotifyCollectionChangedAction.Remove:
+                    foreach (NodeModel oldItem in e.OldItems)
+                        oldItem.PropertyChanged -= OnNodePropertyChanged;
+                    break;
```

Comment:
I'm not touching this in this PR. Not sure why this is in here. It is possible that I moved the method to reorganize things, but it is not in the scope of this PR.